x/gamm/pool-models/stableswap/amm.go

```diff
@@ -291,22 +292,26 @@ func solveCFMMBinarySearch(constantFunction func(osmomath.BigDec, osmomath.BigDe
 	return func(xReserve, yReserve, yIn osmomath.BigDec) osmomath.BigDec {
 		if !xReserve.IsPositive() || !yReserve.IsPositive() || !yIn.IsPositive() {
 			panic("invalid input: reserves and input must be positive")
+		} else if yIn.GTE(yReserve) {
+			panic("cannot input more than pool reserves")
 		}
 		k := constantFunction(xReserve, yReserve)
```

Comment:
What is the purpose of this `constantFunction`? Do we want to replace that with `cfmmConstant` as well?

Reply:
The idea was to keep it more general in case we ever use multiple CFMMs, but you're right in pointing out that this is undermined by hard-coding `cfmmConstant` below. I do think the generality is still important here, so I will go ahead and change the use of `cfmmConstant` below